src/frontend/e2e/pages/upload.ts

```diff
@@ -0,0 +1,96 @@
+import { Page } from "@playwright/test";
+import { faker } from "@faker-js/faker";
+import { SampleData } from "../utils/schemas/sampleData";
+import { UploadData } from "../utils/schemas/uploadData";
+import * as path from 'path';
+
+
+export class UploadSample {
+  public static async uploadFiles(
+    page: Page,
+    uploadData: UploadData,
+  ): Promise<any> {
+    //click upload button
+    await page.locator("a[href$='/upload/step1']").click();
+    await UploadSample.getSampleDataFile(page,uploadData.dataFile);
+    // complete form
+    for(let i = 0; i < uploadData.samples.length; i++){
+      // fill public ID input
+      // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='publicId']`).type(uploadData.samples[i].publicId);
+       // fill collection date input
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='collectionDate']`).type(uploadData.samples[i].collectionDate);
+       // fill search for location input from emergent widget
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::button[@label='Search For Location']`).click();
+      await page.locator("//div[@role='tooltip']/descendant::input").type(uploadData.samples[i].collectionLocation,{delay:150});
+      await page.keyboard.press("ArrowDown");
+      await page.keyboard.press("Enter");
+      // fill sequencing date input
+      // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='sequencingDate']`).type(uploadData.samples[i].sequencingDate);
+       // toggle keep private switch 
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='keepPrivate']`).click();
+    }
+    
+  }
+
+  public static async getSampleDataFile(
+    page: Page,
+    fileName: string
+  ): Promise<void> {
+   
+   await page.setInputFiles(
+      "input[type='file']",
+      path.resolve('e2e/fixtures/sampleData'+fileName)
+    );
+    await page.waitForTimeout(3000);
+    // click continue button
+    await page.locator("a[href$='/upload/step2']").click();
+    await page.waitForSelector("form div[role='table'] > .MuiTableBody-root");
+  }
+
+  // prefix each file name with path. set default if not file provided
+  public static getFullSampleFilePaths(
+    fileNames?: Array<string>
+  ): Array<string> {
+    const files = fileNames !== undefined ? fileNames : ["sampleData.fasta"];
```

Comment:
when would the fileName be undefined?

Reply:
fixed

Reply:
I wanted it to be generic so use can call without specifying the file names, in which it defaults sampleData.fasta. We can obviously pass the default as a parameter, which will be more cleaner